### PR TITLE
Add core documentation about request scenarios

### PR DESCRIPTION
**Pull request by ballinette**

### CODE REVIEW

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+
+![persistence_scenario_details2](./images/request-scenarios/persistence/details2.png)
+
+\#6. A ```Write Worker``` is notified by the internal broker about a new write request.
+
+\#7. The worker calls the ```Write Engine``` service.
+
+\#8. The ```Write Engine``` service performs a request to send the data to the data storage.
+
+\#9. Callback functions are triggered to transmit the response message back to the ```Write Worker```
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: The write engine returns promises, not callbacks :-)

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+
+\#10. The worker sends the feedback message from ElasticSearch to the input task queue (see \#5).
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: Tasks are submitted to workers using the input task queue, but worker's responses are forwarded through the worker *response* queue

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+
+The worker also sends a notification message to the internal broker, to notify subscribing clients, if any (see next section).
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: Here comes version compatibility problems :-)

It's true in the `beta` branch of Kuzzle, but it has been changed in the master branch (or will be in the next develop->master merge).
Now, workers only respond to the response queue. The `notifier` core component does not listen the internal broker anymore, but is invoked explicitly by controllers instead.

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+
+![persistence_overview3](./images/request-scenarios/persistence/overview3.png)
+
+Detailed workflow:
+
+![persistence_scenario_details3](./images/request-scenarios/persistence/details3.png)
+
+\#11. The ```Worker Listener``` that the ```Write Controller``` registered in step \#5. receive a notification from the ```Internal Broker```...
+
+\#12. ... and forwards it back to the ```Router Controller```...
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: It forwards it to the `write` controller, who then forwards it to the `funnel` controller, who in turn forwards it to the `router` controller

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+  },
+  "state": "all"
+}
+```
+
+(see [Filtering Syntax](#filtering-syntax) for more details about filters)
+
+
+The client then listens to the ```<requestId>``` event on the socket.
+Kuzzle will get back to him with a corresponding Room ID using this event.
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: With a corresponding Room ID and a Channel

File: source/includes/core_documentation/_02-request-scenarios.md [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+\#5. The ```Notifier``` broadcasts the message to each related channel to the Websocket and MQ plugins.
+
+\#6. Finally, the plugins send the message to the clients who subscribed to it.
+
+
+##### 3rd step : notify about persisted data
+
+![pubsub_scenario_details2](./images/request-scenarios/pubsub/details3.png)
+
+\#1. The ```Notifier``` component is notified about a new action by the ```Internal Broker``` (see [step \#10. in write scenario](#writing-persistent-data)).
```
Comment (scottinet) [inline source/includes/core_documentation/_02-request-scenarios.md]: Not anymore. Now it's called directly by the controllers handling persistent data.

**Comment by scottinet** [comment]
Nice guide!
Apart from my remarks: +1

**Comment by ballinette** [comment]
gook remarks
The actual doc is just a copy/past from the existing one that was in docs/ folder within kuzzle source.
It not up-to-date in many cases, I'm fixing it ;)

**Comment by ballinette** [comment]
doc updated.

**Comment by benoitvidis** [comment]
+1

**Comment by dbengsch** [comment]
+1

**Pull request merged**